src/Engine/Component/CameraComponents/CameraComponent.cpp

```diff
@@ -5,176 +5,134 @@
+}
 
-    return true;
+glm::vec3 CameraComponent::getLookDir() const {
+    return -w();
+}
 
-    /*
-
-    // TODO : Create loop or function to make this less lines of code
-    // https://www.khanacademy.org/math/linear-algebra/vectors-and-spaces/dot-cross-products/v/point-distance-to-plane
-
-    // Temporary variables for QOL
-    float centerDist;
-    glm::vec3 hypotenuse;
-    
-    // Test if the sphere is completely in the negative side of the far frustum plane
-    hypotenuse = sphere.origin - farPlanePoint;
-    centerDist = glm::length(hypotenuse) *
-        glm::dot(hypotenuse, farPlaneNormal) /
-        glm::length(hypotenuse) / 
-        glm::length(farPlaneNormal);
-    if (centerDist < 0 && -centerDist > sphere.radius)
-        return false;
-
-    // Test if the sphere is completely in the negative side of the near frustum plane
-    hypotenuse = sphere.origin - nearPlanePoint;
-    centerDist = glm::length(hypotenuse) *
-        glm::dot(hypotenuse, nearPlaneNormal) /
-        glm::length(hypotenuse) /
-        glm::length(nearPlaneNormal);
-    if (centerDist < 0 && -centerDist > sphere.radius)
-        return false;
-    
-    // Test if the sphere is completely in the negative side of the top frustum plane
-    hypotenuse = sphere.origin - topPlanePoint;
-    centerDist = glm::length(hypotenuse) *
-        glm::dot(hypotenuse, topPlaneNormal) /
-        glm::length(hypotenuse) /
-        glm::length(topPlaneNormal);
-    if (centerDist < 0 && -centerDist > sphere.radius)
-        return false;
-    
-    // Test if the sphere is completely in the negative side of the bottom frustum plane
-    hypotenuse = sphere.origin - bottomPlanePoint;
-    centerDist = glm::length(hypotenuse) *
-        glm::dot(hypotenuse, bottomPlaneNormal) /
-        glm::length(hypotenuse) /
-        glm::length(bottomPlaneNormal);
-    if (centerDist < 0 && -centerDist > sphere.radius)
-        return false;
-
-    // Test if the sphere is completely in the negative side of the left frustum plane
-    hypotenuse = sphere.origin - leftPlanePoint;
-    centerDist = glm::length(hypotenuse) *
-        glm::dot(hypotenuse, leftPlaneNormal) /
-        glm::length(hypotenuse) /
-        glm::length(leftPlaneNormal);
-    if (centerDist < 0 && -centerDist > sphere.radius)
-        return false;
-
-    // Test if the sphere is completely in the negative side of the right frustum plane
-    hypotenuse = sphere.origin - rightPlanePoint;
-    centerDist = glm::length(hypotenuse) *
-        glm::dot(hypotenuse, rightPlaneNormal) /
-        glm::length(hypotenuse) /
-        glm::length(rightPlaneNormal);
-    if (centerDist < 0 && -centerDist > sphere.radius)
-        return false;
+namespace {
 
-    return true;
+float distToPlane(const glm::vec4 & plane, const glm::vec3 & p) {
+    return plane.x * p.x + plane.y * p.y + plane.z * p.z + plane.w;
+}
 
-    */
+}
+
+const bool CameraComponent::sphereInFrustum(const Sphere & sphere) const {
```

Comment:
Good for now - I'll work on moving this to RenderSystem

Reply:
I take that back. Shadow mapping will need this functionality. Leaving this here is fine